### DISCUSSION TITLE
How can we run immediately a scheduler?

### DISCUSSION THREAD

**Discussion by vadympop** [agronholm/apscheduler#840] [Q&A] [answered]
If I want to run the schedule right now without waiting until the next_fire_time, what can I do?
Something like this I'd like to see:
```python
await scheduler.get_job("").do_immediately()
```

**Comment by agronholm** [comment, marked as answer]
On APScheduler 4 (still in alpha), you can: `await scheduler.run_job("my_task_id")`

**Reply by vadympop** [reply]
How to get task_id?

**Reply by agronholm** [reply]
You can pass either a function or a string to `run_job()`. If you've previously configured a task, you would know its ID. Otherwise you can pass a function object, in which case APScheduler automatically computes the default task ID for it.

**Reply by vadympop** [reply]
Okay, thank you
